rskj-core/src/main/java/co/rsk/peg/PegUtils.java

```diff
@@ -37,7 +42,81 @@ public static PegTxType getTransactionTypeUsingPegoutIndex(
         }
     }
 
-    public static boolean isAnyUTXOAmountBelowMinimum(Coin minimumPegInTxValue, BtcTransaction btcTx, Wallet wallet) {
-        return PegUtilsLegacy.isAnyUTXOAmountBelowMinimum(minimumPegInTxValue, btcTx, wallet);
+    /**
+     * Checks if all utxos sent to the <b>fedWallet</b> are equal or above the minimum pegin value.
+     * If no utxo is sent to the <b>fedWallet </b>or there is at least 1 utxo sent to the <b>fedWallet</b> that is below minimum,
+     * this method will return false. It will return true when all utxos sent to the <b>fedWallet</b> are above the minimum pegin value.
+     *
+     * @param minimumPegInTxValue
+     * @param btcTx
+     * @param fedWallet
+     * @param activations
+     * @return
+     */
+    public static boolean allUTXOsToFedAreAboveMinimumPeginValue(Coin minimumPegInTxValue, BtcTransaction btcTx, Wallet fedWallet, ActivationConfig.ForBlock activations) {
+
+        if(!activations.isActive(ConsensusRule.RSKIP379)) {
+            return !PegUtilsLegacy.isAnyUTXOAmountBelowMinimum(minimumPegInTxValue, btcTx, fedWallet);
+        }
+
+        List<TransactionOutput> fedUtxos = btcTx.getWalletOutputs(fedWallet);
+
+        if(fedUtxos.isEmpty()) {
+            return false;
+        }
+
+        return fedUtxos.stream()
+                .allMatch(transactionOutput -> transactionOutput.getValue().compareTo(minimumPegInTxValue) >= 0);
+
+    }
+
+    public static PegTxType getTransactionType(
+        ActivationConfig.ForBlock activations,
+        BridgeStorageProvider provider,
+        BridgeConstants bridgeConstants,
+        Federation activeFederation,
+        Federation retiringFederation,
+        BtcTransaction btcTransaction,
+        long btcTransactionHeight
+    ) {
+
+        List<Federation> liveFeds = new ArrayList<>();
+        liveFeds.add(activeFederation);
+        if (retiringFederation != null){
+            liveFeds.add(retiringFederation);
+        }
+        Wallet liveFederationsWallet = new BridgeBtcWallet(Context.getOrCreate(bridgeConstants.getBtcParams()), liveFeds);
+
+        // TODO: THIS VALUE IS PENDING TO BE DEFINE. PLEASE UPDATE THIS CONSTANT LATER WITH THE CORRECT VALUE.
+        int btcHeightWhenPegoutTxIndexActivates = bridgeConstants.getBtcHeightWhenPegoutTxIndexActivates();
+        int pegoutTxIndexGracePeriodInBtcBlocks = bridgeConstants.getBtc2RskMinimumAcceptableConfirmations() * 5;
```

Comment:
Let's discuss this, but I think we might consider this becoming a constant. 1000 blocks for example. Or even more

Imagine a pegout is created right before the index activates
- 4000 rsk blocks until it's signed plus 100 BTC blocks untils it's registered
- Consider also that the btc height when the index activates is an estimate, we might fall short

This constant could then be used in the tests as well

Reply:
Let's discuss it.